llvm/include/llvm/Support/CommandLine.h

```diff
@@ -1518,11 +1518,20 @@ class opt
       [](const typename ParserClass::parser_data_type &) {};
 };
 
-extern template class opt<unsigned>;
-extern template class opt<int>;
-extern template class opt<std::string>;
-extern template class opt<char>;
-extern template class opt<bool>;
+#if !defined(LLVM_ENABLE_LLVM_EXPORT_ANNOTATIONS) ||                           \
+    !(defined(_MSC_VER) && !defined(__clang__))
```

Comment:
Is it possible to unify the behaviour here across clang and MSVC? At the very least, provide an associated llvm-project issue to track the MSVC behaviour.